queue/retry.go

```diff
@@ -154,20 +154,18 @@ func (rh *basicRetryHandler) waitForJob(ctx context.Context) error {
 					"queue_id": rh.queue.ID(),
 					"job_id":   j.ID(),
 				}))
+				j.AddError(err)
+
+				if err := rh.queue.CompleteRetry(ctx, j); err != nil {
```

Comment:
Can we add a comment here that we are calling `CompleteRetry` to ensure the job is marked as complete since `CompleteAndPut` failed in `handleJob`?

Reply:
Done.